apps/server/src/modules/provisioning/strategy/oidc/service/schulconnex-user-provisioning.service.ts

```diff
@@ -12,28 +14,48 @@ export class SchulconnexUserProvisioningService {
 	constructor(
 		private readonly userService: UserService,
 		private readonly roleService: RoleService,
-		private readonly accountService: AccountService
+		private readonly accountService: AccountService,
+		private readonly logger: Logger
 	) {}
 
 	public async provisionExternalUser(
 		externalUser: ExternalUserDto,
 		systemId: EntityId,
 		schoolId?: string
 	): Promise<UserDO> {
+		const existingUser: UserDO | null = await this.userService.findByExternalId(externalUser.externalId, systemId);
+
+		let isUniqueEmail = true;
+		if (externalUser.email) {
+			isUniqueEmail = await this.userService.isEmailFromExternalSourceUnique(
+				externalUser.email,
+				existingUser?.externalId
+			);
+
+			if (!isUniqueEmail) {
+				this.logger.warning(new EmailAlreadyExistsLoggable(externalUser.email, externalUser.externalId));
+			}
+		}
+
 		let roleRefs: RoleReference[] | undefined;
 		if (externalUser.roles) {
 			const roles: RoleDto[] = await this.roleService.findByNames(externalUser.roles);
 			roleRefs = roles.map((role: RoleDto): RoleReference => new RoleReference({ id: role.id || '', name: role.name }));
 		}
 
-		const existingUser: UserDO | null = await this.userService.findByExternalId(externalUser.externalId, systemId);
 		let user: UserDO;
 		let createNewAccount = false;
 		if (existingUser) {
 			user = existingUser;
+
+			if (!isUniqueEmail) {
```

Comment:
Can you explain this if-block?

Reply:
i refactored it into a ternary, but the logic ist still the same:
if the email is unique, provision (update) the user normally, so: when an email is provided from external source, use that, if not, leave it as it was.
if it is not unique, do not change email.